homeassistant/components/tuya/base.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+"""Tuya Home Assistant Base Device Model."""
+from __future__ import annotations
+
+import asyncio
+
+from tuya_iot import TuyaDevice, TuyaDeviceManager
+
+from .const import DOMAIN
+
+
+class TuyaHaDevice:
+    """Tuya base device."""
+
+    def __init__(self, device: TuyaDevice, device_manager: TuyaDeviceManager):
+        """Init TuyaHaDevice."""
+        super().__init__()
+
+        self.tuya_device = device
+        self.tuya_device_manager = device_manager
+
+        self.loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(self.loop)
```

Comment:
Don't do this. Modern Python is phasing out using loop directly. Anything that needs it will call for it. If you use it to create tasks, use `hass.async_create_task(…)` instead.